src/graph/graph_type.c

```diff
@@ -77,7 +77,7 @@ void _GraphType_LoadNodes(RedisModuleIO *rdb, Graph *g) {
             if(t & SI_NUMERIC) {
                 propValue[i] = SI_DoubleVal(RedisModule_LoadDouble(rdb));
             } else {
-                propValue[i] = SI_StringValC(RedisModule_LoadStringBuffer(rdb, NULL));
+                propValue[i] = SI_StringVal(RedisModule_LoadStringBuffer(rdb, NULL));
```

Comment:
From RedisModuleAPI:

Like RedisModule_LoadString() but returns an heap allocated string that was allocated with RedisModule_Alloc(), and can be resized or freed with RedisModule_Realloc() or RedisModule_Free().

The size of the string is stored at '*lenptr' if not NULL. The returned string is not automatically NULL terminated, it is loaded exactly as it was stored inside the RDB file.

Looking at _GraphType_SaveNode
it seems like you're not storing the NULL delimiter 

RedisModule_SaveStringBuffer(rdb, prop.value.stringval, strlen(prop.value.stringval));

To fix simply make sure we're saving the NULL delimiter when storing node property within the RDB.
RedisModule_SaveStringBuffer(rdb, prop.value.stringval, strlen(prop.value.stringval)+1);